src/main/java/org/assertj/core/api/Assertions.java

```diff
@@ -1139,12 +1139,48 @@ public static AtomicLongArrayAssert assertThat(AtomicLongArray actual) {
    *
    * @param shouldRaiseThrowable The lambda with the code that should raise the exception.
    * @return The captured exception or <code>null</code> if none was raised by the callable.
+   * @see #catchThrowableOfType(ThrowingCallable, Class)
    */
   public static Throwable catchThrowable(ThrowingCallable shouldRaiseThrowable) {
     return AssertionsForClassTypes.catchThrowable(shouldRaiseThrowable);
   }
 
   /**
+   * Allows catching an {@link Throwable} of a specific type more easily when used with Java 8 lambdas.
+   *
+   * <p>
+   * A call is made to {@link #catchThrowable(ThrowingCallable)}, and an assertion is made on the caught {@link Throwable} to ensure it is of the specified type. If
+   * the assertion passes, then the caught {@link Throwable} is cast to the correct subtype before
+   * being returned, making it convenient to perform subtype-specific assertions on the result.
+   * </p>
+   *
+   * <p>
+   * Example:
+   * </p>
+   *
+   * <pre><code class='java'>{@literal @}Test
+   * public void testException() {
+   *   // when
+   *   CustomParseException e = catchThrowableOfType(() -&gt; { throw new CustomParseException("boom!", 1, 5); },
+   *                                                 CustomParseException.class);
+   *
+   *   // then
+   *   assertThat(e).as("message").hasMessageContaining("boom");
+   *   assertThat(e.getLine()).as("line").isEqualTo(1);
+   *   assertThat(e.getColumn()).as("column").isEqualTo(5);
```

Comment:
remove `as` usage, it does not add much value to the example IMO.

Reply:
In practice I find it makes it easier to find the assertion that failed. But if you feel it makes the example too cluttered I'm happy to leave them out.

Reply:
that was my point :+1:

Reply:
Done.